### ISSUE TITLE
If an environment diff looks like json, use a json differ like jd

### ISSUE THREAD

**Issue by blaggacao** [Gabriel439/nix-diff#52]
Sometimes a environment diff is _actual_ plain json, diffed on one big single row.
As a result an entire screen is coloured red/green without really helpful semantics.
Even the `--character-oriented` is not really of great help at times (when the diff is a little bigger).

It might be possible to use  something like https://github.com/josephburnett/jd to make the result more intelligible.

Since these tools are no haskell, a good way would be to optionally write files with old/new environments to disk, it then becomes trivial to do the semantic diffing by hand.

**Comment by blaggacao** [comment]
I was able to copy chunks manually together from the output, but that's rather brittle.

**Comment by Gabriel439** [comment]
Yeah, we run into this same issue at work, too where the diff fares poorly on large JSON files.  The tool can probably auto-detect JSON and use a more intelligent diff for that case

**Comment by blaggacao** [comment]
A colleague just hinted me at https://github.com/jeffkaufman/icdiff -- which I happily forward really just for general knowledge & inspiration in the context of this issue.
